plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java

```diff
@@ -366,7 +367,9 @@ public static IService getService(IServerAttributes server, Connection connectio
 		}
 		String projectName = OpenShiftResourceUniqueId.getProjectName(uniqueId);
 		List<IService> services = connection.getResources(ResourceKind.SERVICE, projectName);
-		return OpenShiftResourceUniqueId.getByUniqueId(uniqueId, services);
+		IService service = OpenShiftResourceUniqueId.getByUniqueId(uniqueId, services);
+		WatchManager.getInstance().startWatch(service.getProject(), connection);
```

Comment:
I double checked and it seems like starting the watcher twice would not harm, a check whether it already exists is in place preventing multiple watchers per project.